### ISSUE TITLE
Ability to use RecBole to produce top-n recommendation using a user's full profile of ratings

### ISSUE THREAD

**Issue by profsmucker** [RUCAIBox/RecBole#1896]
First thanks for your great work in making RecBole available.

My student and I are trying to use RecBole to produce recommendations, but we want to use it to produce recommendations for a user's full set of ratings and not a train/valid/test split.

In other words, we'd like to do the following:

1. Train a model (with a validation set as needed for tuning).
2. Then have the model make top-k ranking prediction for a given user using all of the user's ratings, i.e. 100% of that user's ratings. 

In other words, we don't want to use RecBole to evaluate models, but we instead want to use it to generate recommendations.

The examples provided often use a 80/10/10 train/validation/test split.  We want to train and have the model tune itself with an appropriate train and validation split, but then we want to run the model on 100% of the user's data, i.e. 100/0/0.

What is the best way for us to do this?

Many thanks,

Mark

**Comment by zhengbw0324** [comment]
@profsmucker 
Hello! 
For data split during training, you can adjust it by configuring the `eval_args` parameter. For configuration methods, please refer to our [documentation](https://recbole.io/docs/user_guide/config/evaluation_settings.html).
As for subsequent inference, you can refer to `recbole/utils/case_study.py` for implementation. In the process, the `full_sort_predict` function of the corresponding model is called to obtain the user's scores for all items. It is worth noting that you should not filter the obtained scores through historical interactions in the training set like [here](https://github.com/RUCAIBox/RecBole/blob/1a9dfeca2c6af7833c7e01de7ae73315ab6029e3/recbole/utils/case_study.py#L72C8-L72C21).
